### DISCUSSION TITLE
XFS vs ext4 performance for download/om file

### DISCUSSION THREAD

**Discussion by meb427** [open-meteo/open-meteo#685] [Ideas]
HI Patrick,

I know that openmeteo is very heavy on read / write and learned it the hard way. I had to upgrade my disk drivers to pci 4.0 and high end drivers to catch up. But curious if file system here could help when converting.

I will try to run a benchmark on this.

Aritcle impact on Kafka:
https://blog.allegro.tech/2024/03/kafka-performance-analysis.html

Was not aware that much over the default in Ubuntu ext4 is slow vs XFS. So could be a quick perf tuning grab.

Does the benchmark by the way test disk performance?

**Comment by meb427** [comment]
System 1

![image](https://github.com/open-meteo/open-meteo/assets/128685138/7291b6cc-f733-4a2a-b868-ec80185f83bd)


System 2

![image](https://github.com/open-meteo/open-meteo/assets/128685138/95c09fe3-bb8f-488d-a3bc-48f5b296ff11)

They don't have workload on the background ( ATOM Nuc ) 
Unfortunalty the disks are not exactly the same but both are pci 3.0.

I tried to run gfs 25 download  same time on both

![image](https://github.com/open-meteo/open-meteo/assets/128685138/699d6ee6-da0c-42b9-80d6-06e780538928)

The difference seem not very important.
Also I notice that during conversion the CPU is not fully used ( noticed the same on Ryzen 9) .

The monitoring running in the background didn't show very big load on disks.

Do you have an idea about batch jobs to do for benchmarking?

**Comment by patrick-zippenfenig** [comment]
Hi,
I wouldn't expect any noticeable differences between XFS and EXT4. Note: The `benchmark` command is purely testing single core CPU/memory, no disk IO, no multi-core performance.

I briefly check the article. I do not think it is relevant to Open-Meteo, because Open-Meteo does not keep high ratios of dirty pages. In fact, `msync` is used to flush data to disk after writing 8 MB or so. 

If you would like to test file system performance, you would have to run the API and fetch weather data for random coordinates to reduce the impact of caching. As long as you are not using spinning HDDs I would not expect that the API is throttled by IO, but rather hit CPU/memory limits.

**Reply by meb427** [reply]
My biggest issue is more in data crunching. I feed the CPU not correctly used here. The disks are under pressure even if my monitoring never showing crazy Gbps speed. The cron jobs can trigger a lot of io wait and that kills the OS disk when I was using one to handle all.
I noticed benchmark was using only one core.

Thanks for the clarification.

**Reply by patrick-zippenfenig** [reply]
High disk IO wait and pressure is expect as the conversion process performs spatial to temporal transformations on a large dataset. Large amounts of memory can help to reduce disk IO. I typically use at least 64 GB.

Most downloaders also support concurrent processing with the `--concurrent N` argument. This can however significantly increase IO pressure. I might also consider using the open-data distribution instead of running your own download and conversion chain: https://github.com/open-meteo/open-data
